woocommerce-gateway-ebanx/gateways/class-wc-ebanx-gateway.php

```diff
@@ -350,55 +350,64 @@ public function admin_options()
 	 */
 	public function process_refund($order_id, $amount = null, $reason = '')
 	{
-		$order = wc_get_order($order_id);
-
-		$hash = get_post_meta($order->id, '_ebanx_payment_hash', true);
+		try {
+			$order = wc_get_order($order_id);
 
-		do_action('ebanx_before_process_refund', $order, $hash);
+			$hash = get_post_meta($order->id, '_ebanx_payment_hash', true);
 
-		if (!$order || is_null($amount) || !$hash) {
-			return false;
-		}
+			do_action('ebanx_before_process_refund', $order, $hash);
 
-		if ( empty($reason) ) {
-			$reason = __('No reason specified.', 'woocommerce-gateway-ebanx');
-		}
+			if (!$order || is_null($amount) || !$hash) {
+				return false;
+			}
 
-		$data = array(
-			'hash'        => $hash,
-			'amount'      => $amount,
-			'operation'   => 'request',
-			'description' => $reason,
-		);
+			if ( empty($reason) ) {
+				$reason = __('No reason specified.', 'woocommerce-gateway-ebanx');
+			}
 
-		$config = [
-			'integrationKey' => $this->private_key,
-			'testMode'       => $this->is_sandbox_mode,
-		];
+			$data = array(
+				'hash'        => $hash,
+				'amount'      => $amount,
+				'operation'   => 'request',
+				'description' => $reason,
+			);
 
-		\Ebanx\Config::set($config);
+			$config = array(
+				'integrationKey' => $this->private_key,
+				'testMode'       => $this->is_sandbox_mode,
+			);
 
-		$request = \Ebanx\EBANX::doRefund($data);
+			\Ebanx\Config::set($config);
 
-		if ($request->status !== 'SUCCESS') {
-			do_action('ebanx_process_refund_error', $order, $request);
+			$request = \Ebanx\EBANX::doRefund($data);
 
-			return false;
-		}
+			if ($request->status !== 'SUCCESS') {
+				do_action('ebanx_process_refund_error', $order, $request);
 
-		$order->add_order_note(sprintf(__('EBANX: Refund requested to EBANX %s - Refund ID: %s - Reason: %s.', 'woocommerce-gateway-ebanx'), wc_price($amount), $request->refund->id, $reason));
+				switch ($request->status_code) {
+					case 'BP-REF-7':
+						$message = __('The payment cannot be refunded because it is not confirmed.', 'woocommerce-gateway-ebanx');
+						break;
+					default:
+						$message = $request->status_message;
+				}
 
-		$refunds = current(get_post_meta((int) $order_id, "_ebanx_payment_refunds"));
+				return new WP_Error('ebanx_process_refund_error', $message);
+			}
 
-		$request->refund->wc_refund = current($order->get_refunds());
+			$refunds = $request->payment->refunds;
 
-		$refunds[] = $request->refund;
+			$order->add_order_note(sprintf(__('EBANX: Refund requested. %s - Refund ID: %s - Reason: %s.', 'woocommerce-gateway-ebanx'), wc_price($amount), $request->refund->id, $reason));
 
-		update_post_meta($order->id, "_ebanx_payment_refunds", $refunds);
+			update_post_meta($order_id, '_ebanx_payment_refunds', $refunds);
 
-		do_action('ebanx_after_process_refund', $order, $request, $refunds);
+			do_action('ebanx_after_process_refund', $order, $request, $refunds);
 
-		return true;
+			return true;
+		}
+		catch (Exception $e) {
```

Comment:
Only way I see to fall here is if some add_action hook throw exception...
Is this proposital?